lib/synapse/haproxy.rb

```diff
@@ -655,18 +693,33 @@ def generate_frontend_stanza(watcher, config)
     end
 
     def generate_backend_stanza(watcher, config)
+      backends = {}
+
+      # The ordering here is important.  First we add all the backends in the
+      # disabled state...
+      @seen.fetch(watcher.name, []).each do |backend_name, backend|
+        backends[backend_name] = backend.merge('enabled' => false)
+      end
+
+      # ... and then we overwite any backends that the watchers know about,
+      # setting the enabled state.
+      watcher.backends.each do |backend|
+        backend_name = construct_name(backend)
+        backends[backend_name] = backend.merge('enabled' => true)
+      end
+
       if watcher.backends.empty?
-        log.warn "synapse: no backends found for watcher #{watcher.name}"
+        log.debug "synapse: no backends found for watcher #{watcher.name}"
       end
 
       stanza = [
         "\nbackend #{watcher.name}",
         config.map {|c| "\t#{c}"},
-        watcher.backends.shuffle.map {|backend|
-          backend_name = construct_name(backend)
+        backends.map {|backend_name, backend|
```

Comment:
curious why you got rid of the `shuffle` here; doesn't it helps to distribute the load more evenly? if all consumers have the same ordering for a set of backends and are using the `roundrobin` algorithm, does that mean that all consumers will first contact the first backend for the first request?